GUI.py

```diff
@@ -200,18 +200,28 @@ def playAudio(self):
         if not self.audio.playing:
             #if self.playback_thread is None or not self.playback_thread.is_alive():
             if self.playback_thread is not None and self.playback_thread.is_alive():
-                self.playback_thread.exit()
+                #self.playback_thread.exit()
                 self.audio.stopPlayback() # Ensure stopping previous playback
+                self.playback_thread.join()
+
+                #start a new thread for playback
+            self.playback_thread = threading.Thread(target=self.audio.play)
+            self.playback_thread.start()
+            self.audio.playing = True
+
+            #safely update button states using `after()`
+            self.after(0, lambda: self.pauseButton.configure(state="normal"))
+            self.after(0, lambda: self.playButton.configure(state="disabled"))
                 # Start a new thread if not already playing or if the thread is not alive
                 #self.playback_thread = threading.Thread(target=self.audio.play)
                 #self.playback_thread.start()
```

Comment:
Should remove the commented out code rather than keeping it, makes it more difficult to understand